internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -0,0 +1,356 @@
+}
+
+func (opts *Opts) newProviderSettings() *atlas.ProviderSettings {
+	return &atlas.ProviderSettings{
+		InstanceSizeName:    tier,
+		ProviderName:        tenant,
+		RegionName:          opts.Region,
+		BackingProviderName: opts.Provider,
+	}
+}
+
+func (opts *Opts) askClusterFlags() error {
+	var qs []*survey.Question
+
+	message := "Insert the cluster name"
+	clusterName := opts.ClusterName
+	if clusterName == "" {
+		clusterName = opts.newClusterName()
+		if clusterName != "" {
+			message = fmt.Sprintf("Insert the cluster name [Press Enter to use the auto-generated name '%s']", clusterName)
+		}
+		qs = append(qs, newClusterNameQuestion(clusterName, message))
+	}
+
+	if opts.Provider == "" {
+		qs = append(qs, newClusterProviderQuestion())
+	}
+
+	if err := survey.Ask(qs, opts); err != nil {
+		return err
+	}
+
+	if regionQ := newRegionQuestions(opts.Region, opts.Provider); regionQ != nil {
+		// we call survey.Ask two times because the region question needs opts.Provider to be populated
+		if err := survey.Ask([]*survey.Question{regionQ}, opts); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// askDBUserAccessListFlags allows the user to set required flags by using interactive prompts
+func (opts *Opts) askDBUserAccessListFlags() error {
+	var qs []*survey.Question
+
+	message := "Insert the Username for authenticating to MongoDB"
+	dbUser := opts.DBUsername
+	if dbUser == "" {
+		dbUser = dbUsername()
+		if dbUser != "" {
+			message = fmt.Sprintf("Insert the Username for authenticating to MongoDB [Press Enter to use '%s']", dbUser)
+		}
+
+		qs = append(qs, newDBUsernameQuestion(dbUser, message, opts.validateUniqueUsername))
+	}
+
+	if opts.DBUserPassword == "" {
+		qs = append(qs, newDBUserPasswordQuestion())
+	}
+
+	if len(opts.IPAddresses) == 0 {
+		message = "Insert the IP entry to add to the Access List"
+		publicIP := net.IPAddress()
+		if publicIP != "" {
+			message = fmt.Sprintf("Insert the IP entry to add to the Access List [Press Enter to use your public IP address '%s']", publicIP)
+		}
+		q := newAccessListQuestion(publicIP, message)
+		qs = append(qs, q)
+	}
+
+	if len(qs) > 0 {
+		if err := survey.Ask(qs, opts); err != nil {
+			return err
+		}
+	}
+
+	if len(opts.IPAddresses) == 0 && opts.IPAddress != "" {
+		opts.IPAddresses = []string{opts.IPAddress}
+	}
+
+	if opts.DBUserPassword == "" {
+		// The user wants to auto-generate the password
+		pwd, err := randgen.GenerateRandomBase64String(passwordLength)
+		if err != nil {
+			return err
+		}
+		opts.DBUserPassword = pwd
+	}
+
+	return nil
+}
+
+func (opts *Opts) validateUniqueUsername(val interface{}) error {
+	username, _ := val.(string)
+	dbUser, err := opts.store.DatabaseUser(convert.AdminDB, opts.ConfigProjectID(), username)
+	if err != nil {
+		if !strings.Contains(err.Error(), fmt.Sprintf("No user with username %s exists.", username)) {
+			return err
+		}
+	}
+
+	if dbUser != nil {
+		return errors.New("a user with this username already exists")
+	}
```

Comment:
[nit] you can use modern go error parsing and the atlas error struct to simplify this
```suggestion
	var target *atlas.ErrorResponse
	if err != nil && errors.As(err, &target) {
		if err.StatusCode == "USERNAME_NOT_FOUND" {
			return nil
		}
		return err
	}
	return fmt.Errorf("username %s already exists", username)
```